### PR TITLE
Fixes #1968 : Java to Kotlin and ButterKnife to Viewbinding Migration

### PR DESCRIPTION
**Pull request by Aditya-gupta99**
Fixes https://mifosforge.jira.com/browse/MOBILE-95

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/synccenterdialog/SyncCenterDialogMvpView.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Created by mayankjindal on 10/07/17.
+ */
+interface SyncCenterDialogMvpView : MvpView {
+    fun showUI()
+    fun showSyncingCenter(centerName: String)
+    fun showSyncedFailedCenters(failedCount: Int)
+    fun setGroupSyncProgressBarMax(count: Int)
+    fun updateGroupSyncProgressBar(i: Int)
```
Comment (avinash14022002) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/synccenterdialog/SyncCenterDialogMvpView.kt:13]: @Aditya-gupta99 ,
Change varrible name.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/synccenterdialog/SyncCentersDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        if (binding.btnHide.text == resources.getString(R.string.dialog_action_ok)) {
+            dismissDialog()
+        } else {
+            hideDialog()
+        }
+    }
+
+    override fun showUI() {
+        binding.pbTotalSyncCenter.max = mCenterList!!.size
+        val totalCenters = mCenterList?.size.toString() + resources.getString(R.string.space) +
```
Comment (avinash14022002) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/synccenterdialog/SyncCentersDialogFragment.kt:79]: @Aditya-gupta99 ,
Null safety

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncgroupsdialog/SyncGroupsDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+    private lateinit var binding: DialogFragmentSyncGroupsBinding
+
+    //    @JvmField
+//    @BindView(R.id.tv_sync_title)
+//    var tv_sync_title: TextView? = null
+//
+//    @JvmField
+//    @BindView(R.id.tv_group_name)
+//    var tv_syncing_group_name: TextView? = null
```
Comment (avinash14022002) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncgroupsdialog/SyncGroupsDialogFragment.kt:33]: @Aditya-gupta99 ,
Why commented?

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncsurveysdialog/SyncSurveysDialogMvpView.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * Created by mayankjindal on 19/07/17.
+ */
+interface SyncSurveysDialogMvpView : MvpView {
+    fun showUI()
+    fun updateSurveyList(mSurveyList: List<Survey>)
+    fun showSyncingSurvey(surveyName: String?)
+    fun showSyncedFailedSurveys(failedCount: Int)
+    fun setQuestionSyncProgressBarMax(total: Int)
+    fun setResponseSyncProgressBarMax(total: Int)
```
Comment (avinash14022002) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncsurveysdialog/SyncSurveysDialogMvpView.kt:15]: @Aditya-gupta99 ,
Varrible name

**Review by avinash14022002 — CHANGES_REQUESTED** [review]
@Aditya-gupta99 ,
Please go through the review comments

**Comment by Aditya-gupta99** [comment]
I have done the required changes

**Review by avinash14022002 — APPROVED** [review]

**Pull request merged**